src/operator/tensor/util/node_op_util.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file node_op_util.h
+ * \brief abstraction for commonly nnvm::Node operations.
+ */
+#ifndef MXNET_OPERATOR_TENSOR_UTIL_NODE_OP_UTIL_H_
+#define MXNET_OPERATOR_TENSOR_UTIL_NODE_OP_UTIL_H_
+#include <mxnet/base.h>
+#include <string>
+#include <unordered_map>
+#include "../../elemwise_op_common.h"
+
+namespace mxnet {
+namespace util {
+
+class NodeOp {
+ private:
+    const nnvm::NodePtr &dependent_node;
+
+ public:
+    explicit NodeOp(const nnvm::NodePtr &dependent_node) : dependent_node{dependent_node} {}
+
+    nnvm::NodeEntry mul(const nnvm::NodeEntry &lhs, const nnvm::NodeEntry &rhs) {
+        return nnvm::NodeEntry{mxnet::op::MakeNode("elemwise_mul",
+                                                   dependent_node->attrs.name + "_mul",
+                                                   {lhs, rhs}, nullptr, &dependent_node)};
+    }
+
+    nnvm::NodeEntry mul(const nnvm::NodeEntry &x, double scalar) {
+        const std::unordered_map<std::string, std::string> scalar_dict =
+            {{"scalar", std::to_string(scalar)}};
+        return nnvm::NodeEntry{mxnet::op::MakeNode("_mul_scalar",
+                                                   dependent_node->attrs.name + "_mul_scalar",
+                                                   {x}, &scalar_dict, &dependent_node)};
+    }
+
+    nnvm::NodeEntry mul(double scalar, const nnvm::NodeEntry &x) {
```

Comment:
Why do we need this function? Can we just let the caller to switch order?

Reply:
Sure, we can. I just felt that it was convenient. Should I remove it?